File: src/main/java/org/voegtle/weatherwidget/data/Statistics.java
package org.voegtle.weatherwidget.data;

import java.util.Collection;
import java.util.HashMap;

public class Statistics {

  public enum TimeRange {
    lastHour, today, yesterday, last7days, last30days;

    public static TimeRange fromString(String rangeStr) {
      for (TimeRange range : values()) {
        if (range.toString().equals(rangeStr)) {
          return range;
        }
      }
      return null;
    }
  }

  private String id;

  private HashMap<TimeRange, StatisticsSet> statistics = new HashMap<>();

  public void add(StatisticsSet set) {
    statistics.put(set.getRange(), set);
  }

  public StatisticsSet get(TimeRange range) {
    return statistics.get(range);
  }

  public Collection<StatisticsSet> values() {
    return statistics.values();
  }

  public String getId() {
    return id;
  }

  public void setId(String id) {
    this.id = id;
  }
}

File: app/src/main/java/org/voegtle/weatherwidget/data/Statistics.kt
package org.voegtle.weatherwidget.data

data class Statistics(val id: String, val kind: String) {

  enum class TimeRange {
    lastHour, today, yesterday, last7days, last30days;

    companion object {
      fun fromString(rangeStr: String): TimeRange? {
        return entries.firstOrNull { it.toString() == rangeStr }
      }
    }
  }

  private val statistics = HashMap<Statistics.TimeRange, StatisticsSet>()

  fun add(set: StatisticsSet) {
    statistics.put(set.range, set)
  }

  operator fun get(range: TimeRange): StatisticsSet? {
    return statistics[range]
  }

  fun values(): Collection<StatisticsSet> {
    return statistics.values
  }
}
